### ISSUE TITLE
win 11 server issue (not working)!

### ISSUE THREAD

**Issue by rubysm** [thevindu-w/clip_share_server#6]
Server not working properly in my windows 11

**Comment by thevindu-w** [comment]
Is there any error log or error message? How did you find that the server is not working?

**Comment by thevindu-w** [comment]
@rubysm any update on this? Is it working now?

**Comment by rubysm** [comment]
The server app didn't appear in notifications but it running in background and i try to pair it with app but no servers found, if u need a tester u can contact me

**Comment by thevindu-w** [comment]
> The server app didn't appear in notifications

I assume you are using the latest release version.

> i try to pair it with app but no servers found

This may happen when the firewall blocks the connection. Please check if the app is allowed through the firewall.

Another possible reason can be an anti-virus software blocking the app if you have any.

**Comment by rubysm** [comment]
I do aware of firewall,and i don't run av programme in my pc it may be the server problem and u check it urself by running it in win 11

**Comment by thevindu-w** [comment]
I tried it on Windows 11, and it worked without any problem.

> The server app didn't appear in notifications

It should only create a tray icon. It does not show a pop-up notification balloon.

On some networks, UDP broadcasts and port scanning may be disabled (possibly for security reasons). If you can try with a mobile hotspot, it might work. I think you are aware of public, private, and domain networks where the firewall allow rules should be specified for the correct type of network connection.

If you can check if port 4337 is listening for TCP connections, then the server should be running. (You can use netstat for that)

If you are willing to find the exact cause, I can provide you with an executable with debug mode enabled. It will print more information that may help find the exact issue.

**Comment by rubysm** [comment]
I can provide you with an executable with debug mode enabled. It will print more information that may help find the exact issue.

Yeah sure

**Comment by thevindu-w** [comment]
@rubysm Here is the debug exe file. Since GitHub doesn't allow attaching exe files directly, I have uploaded the zip file which can be extracted.
Debug file: [clip_share_debug.zip](https://github.com/thevindu-w/clip_share_server/files/12700877/clip_share_debug.zip)

You can simply double click to run it. But unlike the release version, this will open a window with logs where you can see what's happening.

**Comment by ovulatingwife** [comment]
@thevindu-w 
![image](https://github.com/thevindu-w/clip_share_server/assets/91519026/a63b09a0-cd03-4b90-9b98-52e894f356c4)

**Comment by thevindu-w** [comment]
@ovulatingwife Which version you are using? 2.4 or 2.5?

**Comment by ovulatingwife** [comment]
@thevindu-w Debug file: [clip_share_debug.zip](https://github.com/thevindu-w/clip_share_server/files/12700877/clip_share_debug.zip)

**Comment by thevindu-w** [comment]
@ovulatingwife  It has that issue because it is based on 2.4. Please use the latest release 2.5 from the releases. I have fixed that in 2.5.

**Comment by ovulatingwife** [comment]
@thevindu-w The 2.5 version I tried first had no interface and no taskbar icon. The debug version I tried later was the picture I just uploaded (I used Google Translate)

**Comment by thevindu-w** [comment]
Here is a debug version of 2.5
[clip_share_2.5_debug.zip](https://github.com/thevindu-w/clip_share_server/files/13060635/clip_share_2.5_debug.zip)

**Comment by ovulatingwife** [comment]
@thevindu-w Sorry for the late reply, debug 2.5 can be used normally.

**Comment by thevindu-w** [comment]
Ok I'll check what's the issue with the release version 2.5.
@ovulatingwife Is the debug version showing the tray icon and are chinese characters sending properly?

**Comment by ovulatingwife** [comment]
@thevindu-w Yes, debug 2.5 has displayed the tray icon normally and can also send Chinese normally. However, when Android sends to Windows, it still needs to be manually clicked to send. Selecting the text and clicking to share has no effect.
![Screenshot_2023_1021_204935](https://github.com/thevindu-w/clip_share_server/assets/91519026/7190ecec-e13f-42d3-a2bb-975b3ac12522)

**Comment by thevindu-w** [comment]
Since the version 2.5 is working on my machine, I couldn't still reproduce the issue on my machine. But I have a guess on what's wrong in 2.5. To check if that's the actual issue, can you please check this version. (if there are any running instances of previous versions, please stop them before running this)
v2.5.1-alpla: [clip_share_2.5.1-alpha.zip](https://github.com/thevindu-w/clip_share_server/files/13060712/clip_share_2.5.1-alpha.zip)
If this works, I'll create a new patch release.

> However, when Android sends to Windows, it still needs to be manually clicked to send. Selecting the text and clicking to share has no effect.

I'm still working on that.

**Comment by ovulatingwife** [comment]
@thevindu-w v2.5.1-alpla可以正常使用

I think you can refer to pasteasy (I have also been using it. This software is no longer maintained, and it does not seem to be open source. After a few days of use, you may not be able to connect and need to reset the software (need to delete C:\Users\name\AppData\Roaming \Pasteasy\pasteasy.cfg), there is another problem that the file cannot be sent, and the Android 10+ clipboard cannot be automatically transferred in the background. You need to select Share to Windows, and there is no solution to this problem. It is caused by the system. To protect privacy, the following is the installation package:

**Comment by ovulatingwife** [comment]
@thevindu-w 
[Pasteasy.zip](https://github.com/thevindu-w/clip_share_server/files/13060763/Pasteasy.zip)

**Comment by thevindu-w** [comment]
> v2.5.1-alpla可以正常使用

Since v2.5.1-alpha is working, the issue I guessed was the actual problem that made it not working. I'll create a new release with that issue fixed.

> I think you can refer to pasteasy

I'll check Pasteasy.

> Android 10+ clipboard cannot be automatically transferred in the background

This is a restriction in Android 10+. Android says that it is a privacy enhancement to prevent background access to the clipboard. Seems there is no solution to this (other than having some floating component always being in the foreground).

**Comment by thevindu-w** [comment]
Released [version 2.5.1](https://github.com/thevindu-w/clip_share_server/releases/v2.5.1) that fixes Windows version not working.
@rubysm Can you please check if version 2.5.1 fixes your issue also? If it works, we can close this issue.

**Comment by rubysm** [comment]
> Released [version 2.5.1](https://github.com/thevindu-w/clip_share_server/releases/v2.5.1) that fixes Windows version not working.
> @rubysm Can you please check if version 2.5.1 fixes your issue also? If it works, we can close this issue.

Sry for late reply, Its working but connection not made.

**Comment by rubysm** [comment]
All done but no connection.

**Comment by thevindu-w** [comment]
Can you check if it accepts TCP connections to port 4337. You can check it using a tool like netcat or equivalent running on the localhost. This is to see if it accepts connections or not.

**Comment by rubysm** [comment]
Its on listen mode from which i saw in firewall,connection not established

**Comment by thevindu-w** [comment]
Is the IP address of the **laptop** correctly specified in the client? Can you try to run a different server (HTTP server or any other) and check if the mobile can connect to it using the same IP (for example, web browser if it is an HTTP server)

**Comment by rubysm** [comment]
I already use many local net programs in my pc

**Comment by thevindu-w** [comment]
> I already use many local net programs in my pc

Can the mobile connect to them?

**Comment by rubysm** [comment]
Yeah its too.

**Comment by thevindu-w** [comment]
If you can check ClipShare actually accept connections from localhost, then we can see if the issue is with the server or if it's with some external factor (like firewall).

**Comment by rubysm** [comment]
Yeah reset

**Comment by thevindu-w** [comment]
> Yeah reset

That means the server **is accepting connections** and it reset the connection because it was HTTP.

Can you try the same with a web browser in mobile?

**Comment by rubysm** [comment]
Not loading

**Comment by thevindu-w** [comment]
> Not loading

Yes, but I wanted to see if it was "connection reset" or "refused to connect". Did you use the correct IP address (I updated my last comment)

**Comment by rubysm** [comment]
Reset appears

**Comment by thevindu-w** [comment]
> Reset appears

That means the server is working and the client can reach it. Can you check in the ClipShare mobile client app settings if the port is set to 4337. (I assume you are using the latest moblie client)

**Comment by rubysm** [comment]
Yeah

**Comment by thevindu-w** [comment]
> Yeah

What is the error message shown in the mobile client when you press "Get" button?

**Comment by rubysm** [comment]
Its working but scan button says no servers.

**Comment by thevindu-w** [comment]
> Its working but scan button says no servers.

That means the scan is not getting a reply. The laptop should allow incomming UDP connections to port 4337 on the  **BROADCAST**  address of the network (NOT the address of the laptop). Can you check this in the firewall settings?

**Comment by rubysm** [comment]
Auto copy host to client ,vice versa works ?
Without visiting app.

**Comment by thevindu-w** [comment]
> Auto copy host to client ,vice versa works ?

I'm still working on that. I'll ping when I create a release that supports it. There are some issues with the restrictions introduced in Android 10+ that prevents background clipboard access (they say that it is a privacy enhancement to prevent background clipboard access)

**Comment by rubysm** [comment]
> > Its working but scan button says no servers.
> 
> That means the scan is not getting a reply. The laptop should allow incomming UDP connections to port 4337 on the  **BROADCAST**  address of the network (NOT the address of the laptop). Can you check this in the firewall settings?

Firewall allows both in and out

**Comment by rubysm** [comment]
> > Auto copy host to client ,vice versa works ?
> 
> I'm still working on that. I'll ping when I create a release that supports it. There are some issues with the restrictions introduced in Android 10+ that prevents background clipboard access (they say that it is a privacy enhancement to prevent background clipboard access)

It can be done with root.

**Comment by thevindu-w** [comment]
> Firewall allows both in and out

It's not the direction, I mean the destination address. For example, if your laptop is 192.168.10.15 and phone is 192.168.10.18 where both are in the subnet 192.168.10.0/24, then the broadcast address will be 192.168.10.255. Also note that scanning uses UDP unlike in Get text functionalities which use TCP.

**Comment by rubysm** [comment]
> > Firewall allows both in and out
> 
> It's not the direction, I mean the destination address. For example, if your laptop is 192.168.10.15 and phone is 192.168.10.18 where both are in the subnet 192.168.10.0/24, then the broadcast address will be 192.168.10.255. Also note that scanning uses UDP unlike in Get text functionalities which use TCP.

Sry its far from my knowledge.

**Comment by thevindu-w** [comment]
> It can be done with root.

Yes, but I don't think many users will like to root there phone. There are some workarounds used in some other apps to sync clipboard. I'm planning to try something like that (may be using a floating element) without rooting.

**Comment by thevindu-w** [comment]
> Sry its far from my knowledge.

It's ok, if your laptop can have a fixed IP always on your network (maybe using a static IP), you don't need the scan option because the client app saves the last used IP and you don't need to re-enter it again next time when you open the app.

**Comment by rubysm** [comment]
U can use this as a option for root

**Comment by rubysm** [comment]
It consumes more memory without root

**Comment by thevindu-w** [comment]
> It consumes more memory without root

Is there any reason why it uses significantly more memory without root? I haven't observed any noticeable memory usage in the client.

**Comment by rubysm** [comment]
It consumes more memory without root
It may not stable

**Comment by thevindu-w** [comment]
> U can use this as a option for root

I'll keep this in mind when I'm implementing the background sync option.

**Comment by rubysm** [comment]
> > It consumes more memory without root
> 
> Is there any reason why it uses significantly more memory without root? I haven't observed any noticeable memory usage in the client.

That wrkaroud u e saying which isnt stable

**Comment by rubysm** [comment]
https://www.f-droid.org/packages/com.kpstv.xclipper/

**Comment by rubysm** [comment]
> https://www.f-droid.org/packages/com.kpstv.xclipper/

Not stable when i used it

**Comment by thevindu-w** [comment]
> That wrkaroud u e saying which isnt stable

Yes. Can you suggest a better workaround without root? This is currently the best option I found without root (I can't force users to use root. So it's better to have a non-root option also)

**Comment by thevindu-w** [comment]
> Shizuku non root

I'll check that. I have not yet decided which method to use for background sync, so if this method looks better, I'll use that. Thank you for your suggestion.

**Comment by thevindu-w** [comment]
Since the main problem of ClipShare not working in this issue is solved (at least without scanning), can we close this issue now?

**Comment by rubysm** [comment]
https://www.f-droid.org/packages/org.kde.kdeconnect_tp/
This wrkaround

**Comment by thevindu-w** [comment]
> https://www.f-droid.org/packages/org.kde.kdeconnect_tp/

Isn't this use the floating element as the workaround? When I went through their code, I saw something like that.

**Comment by thevindu-w** [comment]
Ok, I have to check more about that anyway before implementing. I didn't go through the entire code yet.

**Comment by thevindu-w** [comment]
> However, when Android sends to Windows, it still needs to be manually clicked to send. Selecting the text and clicking to share has no effect.

@ovulatingwife This feature is now available in the latest release. You can enable auto-send text and/or files from settings. Once enabled, selecting the text and clicking to share with ClipShare will send it without pressing the send button.
